src/gestures.c

```diff
@@ -625,9 +601,20 @@ static int trigger_swipe_unsafe(struct Gestures* gs,
 	if (!cfg->scroll_smooth && cfg_swipe->dist <= 0)
 		return 0;
 
-	dir = get_swipe_dir_n(touches, touches_count);
-	dir = trig_generalize(dir);
+	dir = trig_generalize(get_swipe_dir_n(touches, touches_count));
 	button = get_button_for_dir(cfg_swipe, dir);
+
+	// Update gesture move_dist before aborting
+	// If we abort because button == -1,
+	// there can still be a little bit of movement.
+	// This means that if a movement was slow or in a strange direction,
+	// this will not be added to move_dist.
+	// To prevent that, we do this calculation before aborting.
+	get_swipe_avg_xy(touches, touches_count, &avg_move_x, &avg_move_y);
+	// hypot(1/n * (x0 + ... + xn); 1/n * (y0 + ... + yn)) <=> 1/n * hypot(x0 + ... + xn; y0 + ... + yn)
+	dist = hypot(avg_move_x, avg_move_y);
+	gs->move_dist += ABSVAL(dist);
+
 	if(button == -1){
```

Comment:
~When dir == TR_NONE then button == -1, so after this check dir is always correct.~
It will be TR_NONE only when direction doesn't matter for given gesture: all buttons are set to same value no matter direction. Otherwise button will be -1 for TR_NONE direction.

Reply:
Can't you simply return here if dir == TR_NONE?

Reply:
Probably, I may have been trying to reduce the diff linecount back when I was writing this so the current code structure may not be the best. Let me check it later.